README.md

```diff
@@ -1,81 +1,178 @@
+| Actively maintained                                                                                                                                | ❌                                                       | ✅                                               | ✅                                                      | ✅                                                         |                                              ✅ |
+| Vanilla JavaScript support                                                                                                                | ✅                                                       | ❌                                               | ✅                                                      | ❌                                                         |                                                     ✅ |
+| Fast code generation                                                                                                                      | ✅                                                       | ✅                                               | ❌                                                      | ❌                                                         |                                                     ✅ |
+
+Perhaps the strongest feature of **Protobuf-ES** mentioned above is its generation of idiomatic JavaScript and TypeScript code.  It adopts the best features from the community generators. This means no more clunky getters and setters. You can now use things like the spread operator and make use of the same JavaScript semantics you've grown used to.
+
+For example, given a Protobuf file such as:
+
+```protobuf
+syntax="proto3";
+
+message User {
+  string first_name = 1;
+  string last_name = 2;
+  bool active = 3;
+  User manager = 4;
+  repeated string locations = 5;
+  map<string, string> projects = 6;
+}
 ```
 
+You can initialize your objects conveniently using the `new` operator or passing an initializer object to constructors:
+
+```typescript
+// Using new
+let user = new User();
+
+// Using an object in the constructor
+user = new User({
+  firstName: "Homer",
+  lastName: "Simpson",
+  manager: {  // you can simply pass an initializer object for this nested message field
+      firstName: "Montgomery",
+      lastName: "Burns"
+  },
+});
+```
+
+In addition, you can use direct property access and work with the same familiar JavaScript syntax you're used to.  You won't get confusing methods like `getLocationsList`, `setLocationsList`, `getProjectsMap`, and `clearProjectsMap`.  You won't have to access nested messages by doing things like `msg.getManager().getLocationsList()`.  For example:
+
+```typescript
+user.active = true;
+user.manager.firstName = "Monty";
```

Comment:
Should it be? It's not marked as optional, so it should be deserialized with the default value? I admit I have no experience with `Protobuf-ES`, so I'm not super familiar with the types it emits. Maybe we should show the output (i.e., the `User` class)?

aside: This will only fail to compile if `strict: true`, but you'd be crazy not to have that on now (sadly it's still not the default).

Reply:
It's a message field, and they are always optional.